packages/core/src/node/messaging/ipc-protocol.ts

```diff
@@ -45,13 +45,26 @@ export function checkParentAlive(): void {
 
 export const ipcEntryPoint = process.env[THEIA_ENTRY_POINT];
 
+const exclusion = ['THEIA_*'];
 export function createIpcEnv(options?: {
     entryPoint?: string
     env?: NodeJS.ProcessEnv
 }): NodeJS.ProcessEnv {
-    const op = Object.assign({}, options);
-    const childEnv = Object.assign({}, op.env);
+    const childEnv: NodeJS.ProcessEnv = {};
+
+    if (options && options.env) {
+        for (const excl of exclusion) {
+            const regexp = new RegExp(excl);
+            for (const key in options.env) {
+                if (!regexp.test(key)) {
+                    childEnv[key] = options.env[key];
+                }
+            }
+        }
+    }
```

Comment:
just a FYI, it looks like many lines to filter something from an object. Looks like we could reduce code with map or filter